### PR TITLE
UofT-DSI | Deep Learning - Assignment 1

### PR DESCRIPTION
**Pull request by sehroz**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)

Implemented a CNN model for CIFAR-100 classification with baseline and enhanced versions.

## What did you learn from the changes you have made?

Adding dropout layers significantly improved model performance from 28% to 41% accuracy.

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?

Considered using batch normalization instead of dropout but stuck with dropout as covered in labs.

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?

Initial model was overfitting but solved it by implementing dropout layers from Lab 3.

## How were these changes tested?

Tested by training both models for 30 epochs and comparing validation accuracies.

## A reference to a related issue in your repository (if applicable)

N/A

## Checklist
- [x] I can confirm that my changes are working as intended

### CODE REVIEW

**Review by tianyi21 — APPROVED** [review]
Hi Sehroz!

Thank you for submitting your DL assignments. Your assignment 1 is now complete.

The following comment(s) is for your reference, and there is no need to revise and resubmit your assignment:
* I saw you provided a thorough and complete discussion in this assignment. I believe you have a clear understanding of the problem you're working on and the design choices you have made, but please kindly remember to answer those short open-ended questions hidden between questions (e.g. Task 2b) in future assignments.

Thanks,
Tianyi [LS]
